query-algebrizer/src/clauses/mod.rs

```diff
@@ -329,6 +329,7 @@ impl ConjoiningClauses {
             value_bindings: self.value_bindings.with_intersected_keys(&vars),
             known_types: self.known_types.with_intersected_keys(&vars),
             extracted_types: self.extracted_types.with_intersected_keys(&vars),
+            column_bindings: self.column_bindings.with_intersected_keys(&vars),
```

Comment:
I don't think this will necessarily do what we want in all cases. In particular it'll generate the expanded column comparisons (`datoms00.e = datoms01.e AND datoms00.e = datoms02.v AND…`) within every child CC.

Give this more thought?